lib/models.js

```diff
@@ -1612,9 +1948,44 @@ class ElementMapping {
 
     return clone;
   }
+
+  toJSON() {
+    let fieldMappings = this.rulesFilter.field._rules.reduce((dict, m) => {      
+      var paths = m.sourcePath.map(p => p.fqn).slice();
+      var currDict = dict;
+      while (paths.length > 0) {
+          let key = paths.shift();
+          if (!currDict[key]) {
+              currDict[key] = {};
+          }
+          currDict = currDict[key]
+      }
+      currDict["target"] = m.target;
+
+      return dict;
+    }, {})
+
+    let out = {
+      "name": this.identifier.name,
+      "namespace": this.identifier.namespace,
+      "fqn": this.identifier.fqn,
+      "targetSpec": this.targetSpec,
+      "targetItem": this.targetItem,
+      "mappings": {
+        "fieldMapping": this.rulesFilter.field._rules.map(m => m.toJSON()), //TODO make hierarchial
+        "cardMapping": this.rulesFilter.cardinality._rules.map(m => m.toJSON()),
+        "fixedValue": this.rulesFilter.fixedValue._rules.map(m => m.toJSON())
```

Comment:
Why `fixedValue`, not `fixedValueMapping`?  For consistency, these should all have the `Mapping` postfix, or all _not_ have the `Mapping` postfix.

Reply:
+1